File: 03- Subprogramas y recursividad/subprogramas/java/excepciones/ExcepcionConFinally.java
/**
 * Descripción: Manejo de excepciones con finally.
 * Autor: Santiago Nicolás De la mora Núñez
 * Fecha: 25/02/2026
 */

public class ExcepcionConFinally {
    public static void main(String[] args) {
        int x = 10, y = 0, z = 0;

        try {
            z = x / y;

            System.out.println(z);
        } catch (ArithmeticException e) {
            System.out.println("Error: división entre 0");
        } finally {
            System.out.println("Esto siempre se ejecuta");
        }

        System.out.println("Fin del programa");
    }
}

File: 03- Subprogramas y recursividad/subprogramas/python/excepciones/excepcion_con_finally.py
"""
Descripción: Manejo de excepciones con finally.
Autor: Santiago Nicolás De la mora Núñez
Fecha: 10/04/2026
"""

x = 10
y = 0
z = 0

try:
    z = x / y

    print(z)
except ZeroDivisionError:
    print("Error: división entre 0.\n")
finally:
    print("Esto siempre se ejecuta")

print("Fin del programa")